core/templates/services/audio-player.service.ts

```diff
@@ -133,6 +133,11 @@ export class AudioPlayerService {
     this.setCurrentTime(0);
     this._currentTrack.stop();
     this._stopIntervalSubject.next();
+
+    this._currentTrack = null;
+    this._currentTrackFilename = null;
+    this._lastPauseOrSeekPos = null;
```

Comment:
is it expected to set `_lastPauseOrSeekPos=0` online 132 and `_lastPauseOrSeekPos=null` on the line 139?

Reply:
Good call. I left that there to spark discussion about whether it was ok to change Stop's functionality. But it seems like people are ok with it so I'll get rid of that line and the other two below it, so Stop will fully mimic the "on end" callback 👍

Reply:
Actually it turns out we do need that, otherwise it continues to play while the card has switched over

Reply:
Do we know the logic? If it's not obvious, can you please leave a code comment explaining why not to remove those lines?

Reply:
done